src/Tools/BuildValidator/Util.cs

```diff
@@ -38,6 +41,48 @@ internal static class Util
             }
         }
 
+        internal static bool IsReferenceAssembly(this PEReader peReader)
+        {
+            var reader = peReader.GetMetadataReader();
+            foreach (var attributeHandle in reader.GetCustomAttributes(Handle.AssemblyDefinition))
+            {
+                var attribute = reader.GetCustomAttribute(attributeHandle);
+                if (GetAttributeFullName(reader, attribute) == "System.Runtime.CompilerServices.ReferenceAssemblyAttribute")
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        internal static string GetAttributeFullName(MetadataReader reader, CustomAttribute attribute)
```

Comment:
@tmat I worked out this way of turning a `CustomAttribute` into a full type name but I'm unsure if this is complete and / or the correct approach. Is there somewhere else we do this that I can look at for a better way to do this?

Reply:
Looks right to me